pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -449,24 +449,6 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 				assert.Nil(t, deprovision, "expected no deprovision request")
 			},
 		},
-		{
```

Comment:
Rather than removing this test, can we change the expected result to prove that no deprovision is created?

Reply:
...i.e. it would look like the preceding case, just with `cd.Spec.Installed = false`

Reply:
Since `cd.Spec.Installed` does not have any effect on preserveOnDelete functionality anymore, do we really need this test? We can remove `cd.Spec.Installed = true` from previous test and it will still pass.

Reply:
IMHO yes. The point of UT is not only to make sure the code does what we think; it's also to catch regressions introduced by future code changes. If in the future someone reintroduces logic around `Installed = true`, we want this test to fail.

Happy to be overridden, though. @dgoodwin was [just saying](https://coreos.slack.com/archives/CE3ETN3J8/p1623327974171300) how bloated this suite is...

Reply:
Seems like a good idea to me, lets go for it and keep a test to ensure nobody breaks this in the future.

Reply:
sounds good